### PR TITLE
Healthcheck package

### PR DESCRIPTION
**Pull request by kwizzn**

### CODE REVIEW

File: packages/healthcheck/index.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  function checkServices() {
+    return Promise.all(
+      Object.keys(healthChecks).map(async name => {
+        let dep = healthChecks[name];
+
+        if (typeof dep === 'string') {
+          dep = healthChecks[name] = { method: 'GET', url: dep };
+        }
+        dep.options = Object.assign({ json: true }, dep.options);
+        dep.options.body = { foo: 'bar' };
```
Comment (Sharaal) [inline packages/healthcheck/index.js]: Some missing testdata with foo: 'bar'?

File: packages/healthcheck/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Usage
+
+```
+const healthChecks = {
+  service1: 'https://service1.com/ping',
+  service2: { method: 'POST', url: 'https://service2.com/ping' },
+  service3: { url: 'https://service3.com/ping', options: { headers: { 'Authorization': `Basic dXNlcjpwYXNz` } } },
+};
+
+app.use(require('@maxdome/healthcheck')({ healthChecks }));
```
Comment (Sharaal) [inline packages/healthcheck/README.md]: The router defines "/" and "/details", would it not better to use it with a prefix like "/healthcheck" in the example?

File: packages/healthcheck/index.js (hunk 1 above)
Comment (kwizzn) [reply packages/healthcheck/index.js]: No, that got committed by accident. :)

File: packages/healthcheck/README.md (hunk 2 above)
Comment (kwizzn) [reply packages/healthcheck/README.md]: You're right. I was about to use it like so:
```javascript
app.use('/health', require('@maxdome/healthcheck')({ healthChecks }));
```

I will update `README.md` but we can as well force `/health` and `/health/details` in the package.

File: packages/healthcheck/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        };
+        dep.status = res.statusCode || res.status || 502;
+      })
+    )
+      .then(() => results)
+      .catch(() => results);
+  }
+
+  function hasErrors(result) {
+    return !!Object.values(result).find(val => val.status >= 400);
```
Comment (fabsrc) [inline packages/healthcheck/index.js]: If you use `Array.prototype.some` here, you don't need the `!!` at the beginning

File: packages/healthcheck/index.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        } catch (err) {
+          res = err.response;
+        }
+
+        const connOptions = Object.getOwnPropertySymbols(res.req.connection).find(
+          item => item.toString() === 'Symbol(connect-options)'
+        );
+        dep.request = {
+          body: dep.options.body,
+          headers: res.req.connection[connOptions].headers,
```
Comment (fabsrc) [inline packages/healthcheck/index.js]: Conceptional question: why do we need the headers at all? The Basic Auth header will be displayed in "plaintext" for example. Is this wanted?

**Review by fabsrc — APPROVED** [review]

**Pull request merged**